src/dropsgear.ts

```diff
@@ -0,0 +1,376 @@
+
+  const sporadicMeatBonus = (40 * 0.125 * (equipMode === "barf" ? baseMeat : 0)) / 100;
+  const averageFlowerValue =
+    getSaleValue(
+      ...$items`tin magnolia, upsy daisy, lesser grodulated violet, half-orchid, begpwnia`
+    ) * Math.max(0.01, 0.5 - get("_mayflowerDrops") * 0.11);
+  return new Map<Item, number>([
+    [
+      $item`Mayflower bouquet`,
+      (get("_mayflowerDrops") < 10 ? averageFlowerValue : 0) + sporadicMeatBonus,
+    ],
+  ]);
+}
+
+export function bonusGear(equipMode: BonusEquipMode): Map<Item, number> {
+  return new Map<Item, number>([
+    ...cheeses(equipMode === "embezzler"),
+    ...(equipMode === "embezzler" ? pantsgiving() : []),
+    ...shavingBonus(),
+    ...mafiaThumbRing(equipMode),
+    ...luckyGoldRing(equipMode),
+    ...mrCheengsSpectacles(),
+    ...mrScreegesSpectacles(),
+    ...pantogramPants(),
+    ...bagOfManyConfections(),
+    ...snowSuit(equipMode),
+    ...mayflowerBouquet(equipMode),
+  ]);
+}
+
+export function bestBjornalike(existingForceEquips: Item[]): Item | undefined {
+  const bjornalikes = $items`Buddy Bjorn, Crown of Thrones`;
+  const slots = bjornalikes
+    .map((bjornalike) => toSlot(bjornalike))
+    .filter((slot) => !existingForceEquips.some((equipment) => toSlot(equipment) === slot));
+  if (!slots.length) return undefined;
+  if (slots.length < 2 || bjornalikes.some((thing) => !have(thing))) {
+    return bjornalikes.find((thing) => have(thing) && slots.includes(toSlot(thing)));
+  }
+
+  const hasStrongLep = leprechaunMultiplier(meatFamiliar()) >= 2;
+  const goodRobortHats = $items`crumpled felt fedora`;
+  if (myClass() === $class`Turtle Tamer`) goodRobortHats.push($item`warbear foil hat`);
+  if (numericModifier($item`shining star cap`, "Familiar Weight") === 10)
+    goodRobortHats.push($item`shining star cap`);
+  if (have($item`carpe`) && (!hasStrongLep || !goodRobortHats.some((hat) => have(hat)))) {
+    return $item`Crown of Thrones`;
+  }
+  return $item`Buddy Bjorn`;
+}
+
+function shavingBonus(): Map<Item, number> {
+  if (!have($item`Daylight Shavings Helmet`)) return new Map();
+  if (
+    $effects`Barbell Moustache, Cowboy Stache, Friendly Chops, Grizzly Beard, Gull-Wing Moustache, Musician's Musician's Moustache, Pointy Wizard Beard, Space Warlord's Beard, Spectacle Moustache, Surrealist's Moustache, Toiletbrush Moustache`.some(
+      (effect) => have(effect)
+    )
+  ) {
+    return new Map();
+  }
+
+  const bonusValue = (baseMeat * 100 + 72 * 50) / 100;
+  return new Map<Item, number>([[$item`Daylight Shavings Helmet`, bonusValue]]);
+}
+
+/**
+ * Calculates whether we expect to be wearing the thumb ring for most of the farming day.
+ * This is used in functions that leverage projected turns; for instance, calculating the
+ * number of turns of sweet synthesis required in our diet calcs or potion costs.
+ * @returns boolean of whether we expect to be wearing the thumb ring for much of the day
+ */
+export function usingThumbRing(): boolean {
```

Comment:
this function should reflect if we are in noBarf mode or not

Reply:
in nobarf, we don't care about long-term thumb ring use because our expectedTurns() is just embezzlerCount()